test/tests/functional/pbs_mom_hook_sync.py

```diff
@@ -0,0 +1,365 @@
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+class TestMomHookSync(TestFunctional):
+
+    """
+    This test suite tests to make sure a hook does not disappear in
+    a series of hook event change from mom hook to server hook and
+    then back to a mom hook. This is a good exercise to make sure
+    hook updates are not lost even when mom is stopped, killed, and
+    restarted during hook event changes.
+    """
+
+    def setUp(self):
+
+        if len(self.moms) != 2:
+            self.skip_test(reason="need 2 mom hosts: -p moms=<m1>:<m2>")
+        TestFunctional.setUp(self)
+
+        self.momA = self.moms.values()[0]
+        self.momB = self.moms.values()[1]
+        self.momA.delete_vnode_defs()
+        self.momB.delete_vnode_defs()
+
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+
+        rc = self.server.manager(MGR_CMD_DELETE, NODE, None, "")
+        self.assertEqual(rc, 0)
+
+        rc = self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA)
+        self.assertEqual(rc, 0)
+
+        rc = self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostB)
+        self.assertEqual(rc, 0)
+
+        self.hook_name = "cpufreq"
+        hook_body = "import pbs\n"
+        a = {'event': 'execjob_begin', 'enabled': 'True'}
+        self.server.create_import_hook(self.hook_name, a, hook_body)
+
+        hook_config = """{
+    "apple"         : "pears",
+    "banana"         : "cucumbers"
+}
+"""
+        (fd, fn) = self.du.mkstemp()
+        os.write(fd, hook_config)
+        os.close(fd)
+        a = {'content-type': 'application/x-config',
+             'content-encoding': 'default',
+             'input-file': fn}
+        self.server.manager(MGR_CMD_IMPORT, HOOK, a, self.hook_name)
+        os.remove(fn)
+
+        self.server.log_match(
+            'successfully sent hook file.*cpufreq.HK ' +
+            'to %s.*' % self.momA.hostname,
+            max_attempts=10, regexp=True)
+
+        self.server.log_match(
+            'successfully sent hook file.*cpufreq.CF ' +
+            'to %s.*' % self.momA.hostname,
+            max_attempts=10, regexp=True)
+
+        self.server.log_match(
+            'successfully sent hook file.*cpufreq.PY ' +
+            'to %s.*' % self.momA.hostname,
+            max_attempts=10, regexp=True)
+
+        self.server.log_match(
+            'successfully sent hook file.*cpufreq.HK ' +
+            'to %s.*' % self.momB.hostname,
+            max_attempts=10, regexp=True)
+
+        self.server.log_match(
+            'successfully sent hook file.*cpufreq.CF ' +
+            'to %s.*' % self.momB.hostname,
+            max_attempts=10, regexp=True)
+
+        self.server.log_match(
+            'successfully sent hook file.*cpufreq.PY ' +
+            'to %s.*' % self.momB.hostname,
+            max_attempts=10, regexp=True)
+
```

Comment:
Two many blank lines in this function. Similar calls can be grouped together without blank lines between them. Please fix for other functions as well.